### ISSUE TITLE
Enabling Backend Connection via API Key

### ISSUE THREAD

**Issue by HeinrichPet** [International-Data-Spaces-Association/DataspaceConnectorUI#70]
Allow Backend Connection also via API Key next to the current Basic Auth (User name + password).
See: https://github.com/International-Data-Spaces-Association/DataspaceConnector/issues/364

**Comment by BastianWel** [comment]
Authentication data is added to  backend Connections (Generic Endpoints) via Data Sources:
`PUT ​/api​/endpoints​/{id}​/datasource​/{dataSourceId}`
The Swagger documentation of `POST ​/api​/datasources` is
```
{
  "authentication": {
    "key": "string",
    "value": "string"
  },
  "type": "Database"
}
```
I am not sure how to set API Key for this endpoint and did not find any other documentation on this topic by now.

**Comment by BastianWel** [comment]
API key authentication can currently only be set directly on the artifact. 
DataSource needs to be extended so that the user can save the API key authentication directly with the Backend Connection (Generic Endpoint) just like BasicAuth.

**Comment by BastianWel** [comment]
Current solution: 
Basic auth: set key=user, value=password for data source
API-Key: only set value=api key

Implemented in e58f1db
